src/ipc/server.rs

```diff
@@ -106,29 +108,40 @@ fn on_new_ipc_client(state: &mut State, stream: UnixStream) {
 
 async fn handle_client(ctx: ClientCtx, stream: Async<'_, UnixStream>) -> anyhow::Result<()> {
     let (read, mut write) = stream.split();
-    let mut buf = String::new();
 
-    // Read a single line to allow extensibility in the future to keep reading.
-    BufReader::new(read)
-        .read_line(&mut buf)
-        .await
-        .context("error reading request")?;
+    // note that we can't use the stream json deserializer here
+    // because the stream is asynchronous and the deserializer doesn't support that
+    // https://github.com/serde-rs/json/issues/575
 
-    let reply = process(&ctx, &buf).map_err(|err| {
-        warn!("error processing IPC request: {err:?}");
-        err.to_string()
-    });
+    let mut lines = BufReader::new(read).lines();
 
-    let buf = serde_json::to_vec(&reply).context("error formatting reply")?;
-    write.write_all(&buf).await.context("error writing reply")?;
+    while let Some(line) = lines.next().await {
```

Comment:
Do you mind splitting out the "multiple requests per connection" into a separate PR? I'm not convinced that we need it *right now*, but I would like the rest of this PR in (version, return-error, version checks).

Reply:
i made it no longer handle multiple requests on a connection; most of the inner workings to do so are still present, they're just unused.

Reply:
Thanks. I think the `NiriSocket` struct is good, but I'd further roll back some logic complexity that will only be necessary for multiple requests per connection. I can do it myself if you'd prefer to save on the comments back and forth.